manifests/tools.pp

```diff
@@ -1,8 +1,20 @@
-# Class to ensure qpid tools package
-class qpid::tools {
+# Class to manage qpid tools package
+#
+# @param ensure
+#  Specify 'present' to install qpid-tools package or 'absent' to remove the package and any dependencies
+#
+class qpid::tools(
+  Enum['present', 'absent'] $ensure = 'present',
+) {
+
+  if $qpid::tools::ensure == 'absent' {
+    $_package_ensure = 'purged'
+  } else {
+    $_package_ensure = $qpid::tools::ensure
+  }
```

Comment:
It should be in the local scope. Doesn't this also work?
```suggestion
  if $ensure == 'absent' {
    $_package_ensure = 'purged'
  } else {
    $_package_ensure = $ensure
  }
```

You can also use
```suggestion
  $_package_ensure = bool2str($ensure == 'absent', 'purged', $ensure)
```

That might also be short enough to use directly with an additional variable.

Reply:
Thanks. I'm not sure whether the saved space in the latter example is worth the cost to readability? I'll make an update to keep `if ... { ... } else { ... }` but use local scope as in the former example.

If you feel strongly that it's better code quality to use `bool2str(...)` instead, I think there are other places where we should make that change as well.